### ISSUE TITLE
‹ĥ› is ugly

### ISSUE THREAD

**Issue by ctrlcctrlv** [alerque/libertinus#329]
**Describe the bug**
&lsaquo;ĥ&rsaquo; is <del>wrong</del> ugly.

**Screenshots / logs**
![image](https://user-images.githubusercontent.com/838783/86499539-a4489c00-bd40-11ea-85d5-d9cae81e7e2c.png)

**Additional context**
The first &lsaquo;ĥ&rsaquo; is how Libertinus renders it now. The second &lsaquo;ĥ&rsaquo; is how most fonts render it. The third &lsaquo;ĥ&rsaquo; is an acceptable, if rare, alternative.

&lsaquo;ĥ&rsaquo; is used in Esperanto. I'm the closest thing to a native Esperanto speaker that exists. My Wikipedia username contains an &lsaquo;ĥ&rsaquo;, so I've seen many &lsaquo;ĥ&rsaquo;'s in all kinds of fonts. Libertinus' &lsaquo;ĥ&rsaquo; is wrong.

"But Fred, Esperanto is a made up language, surely we can make up a glyph", some might say. No. Zamenhof was the one who could have chosen Libertinus' &lsaquo;ĥ&rsaquo;, but he didn't. Now Esperanto is old enough that there are ways of doing things and the &lsaquo;ĥ&rsaquo; is clearly wrong.

**Comment by alerque** [comment]
What other language(s) if any use h with a circumflex? Since this isn't a language specific combination it seems like we need some research and precedent from other places it may appear as well before changing it up. The scope of who else besides Esperanto would be affected should be ascertained first. Do you have any knowledge of that?

**Comment by moyogo** [comment]
Ĥ ĥ are also used in Bassari language in Senegal.
This is what ĥ looks like in the law that sets the official Bassari orthographic rules in Senegal:
<img width="955" alt="image" src="https://user-images.githubusercontent.com/1923747/86513471-7c217180-be02-11ea-9795-b9d5c69a787b.png">
I don’t think the form is normative, it just happens to be the form in the font used in the _Journal officiel de la République du Sénégal_ in which laws are published.

That said, the second and third forms in the OP screenshot are the most common forms, generally speaking.

**Comment by moyogo** [comment]
I guess if one digs enough, examples of ĥ with the circumflex low on the right of the ascender can be found.

Here one I found after going through a dozen of Zamenhof’s early Esperanto books on https://www.onb.ac.at/bibliothek/sammlungen/plansprachen/digitale-medien/ludwik-l-zamenhof/ (namely [Zamenhof, Ludwik Lazar (1889), _Dr. Esperanto’s international language_, Warszawa](http://anno.onb.ac.at/cgi-content/anno-buch?aid=100059), [page 35](http://anno.onb.ac.at/cgi-content/anno-buch?apm=0&aid=100059&teil=0203&seite=00000035&zoom=2)):
<img width="803" alt="image" src="https://user-images.githubusercontent.com/1923747/86513801-04087b00-be05-11ea-9f1f-ca6e166c7600.png">

This however seems to be anecdotal as the other two forms are much more common.

**Comment by ctrlcctrlv** [comment]
@moyogo That's interesting. The ŭ in use there is also very strange indeed. I would caution you though, just because something has happened once in history doesn't make it right. It's certainly news to me that such an old source has this ĥ, but I maintain that this is just an eccentricity of a single book publisher and not representative of modern Esperanto usage.

**Comment by georgd** [comment]
There's no right or wrong on any of the three forms. They can all be seen throughout the history of Esperanto typesetting, though admittedly the present version is the least common. I'd generally refrain from categorising in right and wrong upon such questions as this is about design which is about aesthetics, a highly subjective matter. It'd be wrong if it could too easily be mistaken for a different glyph but that's not the case here.

**Comment by ctrlcctrlv** [comment]
I very much respect you @georgd and love your EB Garamond font but it looks wrong to me. I've owned books in Esperanto, contributed to Esperanto Wikipedia, et cetera. I'm not sure why the weight of my experience is being discounted by a single scanned page.

Now, if you just object to the word "wrong", fine...? Given it happened once, and early too, perhaps it's a rare alternative. So, I'll edit my title...

**Comment by georgd** [comment]
@ctrlcctrlv thank you.

The categorisation as "wrong" is indeed in the foreground of my comment. (Just for the record: in my eye this variant is not ugly – I even find it more pleasing than the more usual variants.)

About the facts: you might visit the Esperanto Museum here in Vienna (https://www.onb.ac.at/eo/museen/esperantomuzeo) and find that the incriminated variant hasn’t been used only once. Yet I do confirm that it’s really rare.

Usually I wouldn’t, but in this case I do recommend to reconsider the design choice. As Esperanto is usually acquired as second/third/ ... language and reading practice in it is mostly not as advanced as in the first language, design variations like this might affect readability, especially in a text font where the glyphs shouldn’t stand out positively or negatively. That’s much less the case for ‘more established’ glyphs like Ö or ç where lots of variants don’t pose a problem.

**Comment by alerque** [comment]
@ctrlcctrlv Nobody is discounting the weight of your experience or even the weight of your opinion — but it doesn't make you automatically right. In fact I think it's pretty clear that your initial argument was an overstatement of your case and also overreached a bit. My first comment addressed the over-reach (by asking about other languages even though your report made it sound like Esperanto was the only factor involved) but I will also say that the overstatement in your choice of words and tone is not endearing your argument. I will still consider this, but please back of from the dogmatic assertion of right and wrong in regard to something that at least _can_ be a matter of style and preference. It might be best for the Esperanto ecosystem to not rock the boat and stick as closely as possible to the most common design choices, but before we go there lets actually understand what decision needs to be made.

A non-exhaustive set of questions I would expect to have answered before changing this:

* What other languages are effected?
* Why does *Pragmatica Esperanto* (one of three fonts showcasing Esperanto's h-circumflex on Wikipedia) also place the circumflex above the shoulder / beside the ascender instead of above the height? If an Esperanto focused font made that choice, it likely isn't 100% wrong and might even be good.
* What other fonts are there that made this choice and is there any paper trail on their reasoning?
* Why did one of Zamenhof's early books go to press with this design? Because they didn't have a choice? Because the author liked it that way? Did he later express a reason for a later change or is this something he still accepts?
* What other letters use similar diacritic positioning in Libertinus right now?
* What if any languages are affected by stacked diacritics in this position?

**Comment by alerque** [comment]
@moyogo Thanks for the references—both good finds.

The reference for usage in Bassari should probably be added to [Wikipedia](https://en.wikipedia.org/wiki/%C4%A4#Other_uses)! How did you find that anyway?

**Comment by georgd** [comment]
Scriptsource lists it as h with tilde (https://scriptsource.org/cms/scripts/page.php?item_id=wrSys_detail_sym&key=bsc-Latn-SN) so, what @moyogo found is perhaps a stopgap as ĥ being available whereas proper support for arbitrary letter/mark combinations is still rare: h̃

**Comment by fitojb** [comment]
The fact that a lot of fonts place the circumflex in ⟨ĥ⟩ in the mathematical middle, not attached to anything and floating in mid-air, is so disappointing. Please don’t use that in Libertinus.

**Comment by moyogo** [comment]
@fitojb It’s not just Arial.

At a quick glance:
- form 1 (low, next to ascender):
  - Libertinus Serif
  - Pragmatica Esperanto
- form 2 (centered above the whole glyph):
  - Brill
  - STIX2 Text
  - Segoe UI
  - Lucida Grande
  - Libertinus Sans
  - Noto Serif
  - Source Sans Pro
  - Source Serif Pro
- form 3 (centered above ascender):
  - Andika
  - Charis SIL
  - Doulos SIL
  - Gentium Plus
  - Castoro
  - Sylfaen
  - SF Pro
  - Noto Sans

I doubt this trend is not reflected in Esperanto documents published in the past couple of decades.

**Comment by moyogo** [comment]
The point I should have clarified when I provided the example with the form matching the current design is that it was one out of the dozen I went through. So, while it does occur, it is not the most common, even in the early days of Esperanto.

**Comment by alerque** [comment]
The [Esperanto language Wikipedia page for h-circumflex](https://eo.wikipedia.org/wiki/%C4%A4) states (Google Translation):

> The use of a roof-shaped accent, the hat, to change the grapheme H into Ĥ is an invention of L. L. Zamenhof. Its lower case has at least three typographic variations: with the hat above the bar of the h, with the hat above the arch of the h ("on the knee"), or with the hat centrally above both.

No mention or even suggestion is made that would lead me to believe the form is considered *wrong* en Esperanto.

On that page besides the sample showing all three, the other sample uses *Linux Libertine O* and evidences the position this font apparently inherited: lower above the shoulder of the h.

Given the availability of other high quality fonts of both Serif and Sans variants that use each of the other style choices and this font being probably the leading one using the 3rd alternative I think we're probably on good ground preserving this stylistic choice.

The one thing that does actually look wrong to me is that Libertinus Sans should make the same choice as it's Serif cousin.

I realize that's literally the *opposite* outcome being advocated by the OP, but the evidence isn't adding up yet. I'm willing to leave this open for a while in hopes of hearing from more Esperanto speakers.

**Comment by ctrlcctrlv** [comment]
I am sorry if you felt like I oversold. I did not do so intentionally.

To my memory, I first started speaking complete Esperanto sentences at 10.

I had an Esperantist in my family (grandfather), so it's possible I knew quite a few words before then, but I only remember being "quite good at it" at 10.

I was given this book by my grandfather, already well loved and had many of his notes...
<img src="https://user-images.githubusercontent.com/838783/86593493-2eb01c00-bf4a-11ea-9c32-5f9bb83b3de1.png" width="200">

It was destroyed in a household accident. He replaced it for me with the newer edition, since my dad was too poor to and didn't see the point.
<img src="https://user-images.githubusercontent.com/838783/86593514-3bcd0b00-bf4a-11ea-804e-e29e9142f317.png" width="200">

Naturally, I've read the ''Fundamento'' and the ''Unua Libro'', both in PDF form. I am a member of the Universala Esperanto-Asocio, and received its newsletter for years. You've reminded me I need to inform them of my new address.

I've also always been interested in fonts. If I would have seen the Libertinus-style ‹ĥ›, I certainly would have noticed. Depending on when I saw it, I either would have posted on lernu.net forum, ##esperanto on Freenode, or Esperanto Facebook group some joke like...

> **ĈIMOMENTA NOVAĴO!** Ĉu rompita presilo... aŭ, aĵo tre sinistra!? &lsaquo;h&rsaquo;on, ni, intervjuis. Ria mesaĝ': &ldquo;Tuj, relokigu mian ĉapelon tuj, barbaro! Tuj!&rdquo; Xerox, komento demandita antaŭ du tagoj&mdash;ankoraŭ ne.&raquo; 
> BREAKING NEWS! A broken printer...or something more sinster? &lsaquo;h&rsaquo; in an exclusive interview: &lsquo;Put my hat back this instant, you barbarian! Right now!&rsquo; Xerox, asked for comment two days ago; as yet we've received none.&rdquo; 

Really, the jokes write themselves, and this would not be a missed opportunity, especially if I saw it in a print source. Certainly I'd circle it and not think it a computer glitch. :-)

So, if I oversold, it was because I thought I'd seen enough ‹ĥ›'s in my life to know the allowable variation in the glyph. Apparently not.

**Comment by ctrlcctrlv** [comment]
Also, @fitojb, sorry to inform you, I actually find the normative &lsaquo;ĥ&rsaquo; with the _ĉapelo_&zwj;¹ centered, more visually appealing, perhaps because I'm much more used to it. It depends on if we're making type that will look most natural/not strange to Esperanto speakers, or if we're making decorative type that will break design barriers, in my opinion. I think for a book face we should use normative forms that will not surprise readers.

However, seeing as @alerque gave your comment 👀, he seems to agree. I much prefer either form to the current &lsaquo;ĥ&rsaquo;, so, I won't complain beyond this comment if it that's what is implemented.

<sup>¹ Literally, hat. This is really what we call it, I don't know why; Vikipedio [says](https://eo.wikipedia.org/wiki/%C4%88apelo_(supersigno)) we can also call it a _tegmento_, lit. roof, which I find cute!</sup>

**Comment by ctrlcctrlv** [comment]
I also have asked other Esperanto speakers to comment, on Twitter:

https://twitter.com/HW_BEAT_THAT/status/1280127676326371329

>Other #Esperanto speakers are asked to kindly give their opinions on the correct shape of the letter ‹ĥ› here: https://github.com/alerque/libertinus/issues/329

And IRC:

```
09:08:08 josomebody1 saluton
09:08:23 -- josomebody1 is now known as josomebody
09:19:57 copypaste kiu estas la ĝusta formo de la nobla ‹ĥ›o? bv. komentu TIE → https://github.com/alerque/libertinus/issues/329 ← 
         (bv. ne ĉi-kanale, eble ili ne min kredos haha, ĉar ververe ŝajnas ili jam(!!) ne min ŝatas :D) se vi tempon havas. antaŭdankon.
09:20:06 sham2 Saluton al ĉiuj!
09:20:07 copypaste josomebody: ^ :-)
[124] [irc/freenode] 5:##esperanto{142}{+cnt}
```

**Comment by georgd** [comment]
I knew that this was brought up already: https://sourceforge.net/p/linuxlibertine/feature-requests/167

**Comment by alerque** [comment]
Thanks for digging that up @georgd. I kind of figured there must be some back story but I'd only searched through _this_ project's old issues.

For the record, I am not currently inclined to change the design. In fact I'm currently keeping the issue open to track fixing the Sans family to match the current Serif family design. This default design choice could certainly be evaluated if it could be demonstrated that there was a heavy majority of users that preferred a different default, but as long as their is a significant minority that likes our current rendering I think it's okay to be one of two fonts catering to the underdog variant. Evidence suggests it isn't outright wrong — it's a valid if less common atheistic choice. Even if we did change the default we'd need to keep the current rendering as an alternate.

That being said I *might* be willing to accept a contribution which added a non-default alternate with the circumflex anchor above the ascender. I don't think I'm interested in the other variant style at all (floating a mile in the air above the ascender height but centered over the whole glyph).

By the way one of my favorite glyphs in Libertinus Serif is the stylistic alternate h:

```sile
\begin{document}
\font[family=Libertinus Serif,size=48pt,features=+salt,style=Italic]
Wash your hands!
\end{document}
```

![image](https://user-images.githubusercontent.com/173595/86756519-1e339a80-c04b-11ea-92c0-f15ad80e8eb8.png)

**Comment by ctrlcctrlv** [comment]
I guess it depends on what this font is supposed to be.

Is it supposed to be an experimental typeface, where the glyphs are all supposed to be breaking design barriers, or is it supposed to be a text face, engineered to cause the least surprise to readers, because surprising readers slows them down.

I can't decide that for you. I guess I thought it was the latter, but you meant it to be the former. (Well, maybe @khaledhosny didn't mean it that way, but you do.)

**Comment by ctrlcctrlv** [comment]
Oh and by the way, nobody who actually speaks the language likes the current design. 

* Me
* the guy from 2013
* @chemoelectric on Twittter [[1]](https://twitter.com/chemoelectric/status/1280148079484731393)

Have fun with your display typeface, guys.

**Comment by georgd** [comment]
@ctrlcctrlv I’m sorry, there are some misunderstandings here. 

- The "guy from 2013" found the floating version ugly, the ascender-top version the nicest and the over the shoulder version not bad. At the time, Linux Libertine had floating circumflexes on h in all italics and semibold and the shoulder-top only in regular and bold. These haven’t been consolidated yet.
- The design decision was not met by @khaledhosny but by Philipp H. Poll.

> For the record, I am not currently inclined to change the design. In fact I'm currently keeping the issue open to track fixing the Sans family to match the current Serif family design [...]

The thing is, there is no single "current Serif family design". Italic, bold italic, semibold and semibold italic, as well as the whole sans family are showing the "floating" variant, so apparently Philipp himself at a rather early stage already gave up the shoulder-top variant which is only present in regular and bold. @chemoelectric  points to another issue that I wouldn’t have detected on my own: I wouldn’t put the shoulder-top version in a mathematical text.

**Comment by chemoelectric** [comment]
> @moyogo That's interesting. The ŭ in use there is also very strange indeed. I would caution you though, just because something has happened once in history doesn't make it right. It's certainly news to me that such an old source has this ĥ, but I maintain that this is just an eccentricity of a single book publisher and not representative of modern Esperanto u

> I guess if one digs enough, examples of ĥ with the circumflex low on the right of the ascender can be found.
> 
> Here one I found after going through a dozen of Zamenhof’s early Esperanto books on https://www.onb.ac.at/bibliothek/sammlungen/plansprachen/digitale-medien/ludwik-l-zamenhof/ (namely [Zamenhof, Ludwik Lazar (1889), _Dr. Esperanto’s international language_, Warszawa](http://anno.onb.ac.at/cgi-content/anno-buch?aid=100059), [page 35](http://anno.onb.ac.at/cgi-content/anno-buch?apm=0&aid=100059&teil=0203&seite=00000035&zoom=2)):
> <img alt="image" width="803" src="https://user-images.githubusercontent.com/1923747/86513801-04087b00-be05-11ea-9f1f-ca6e166c7600.png">
> 
> This however seems to be anecdotal as the other two forms are much more common.

I would ignore this document. It is too strange; it is in English but has German quote marks and fails to capitalize ‘Scotch’. And it is not an attempt at handsome printing.

As someone else hints at, my taste is dictated largely by what one would expect to see in a mathematical text. But also there are many examples among commercially available fonts and they favor one or the other design with circumflex high. Storm seems generally to favor something midway between the two placements. I suspect the horizontal placement is simply done to taste, as I would do it.

My own feeling is that whatever the notable type producers are doing is what should be considered ‘proper’ Esperanto, because that is what one would see. There is no ‘tradition’ of a national type style. Esperanto style doesn’t even dictate a preferred form of the quote marks--all kinds are used (but extra space as in French printing is discouraged).

**Comment by moyogo** [comment]
Generally, this is more about the fact that Libertinus’ anchors on "bdhk" is not the most common one, and will seem odd to many users, regardless of the language.
The only exception to this is probably Gaelic Irish, where it is quite common, if not more common, to have the dot above next to the ascender instead of above it.

It’s always possible to find recent examples for any variants but the question should really be what is the more appropriate design.
Which is what the original post tried to say.

Personally, I wouldn’t favour the current design unless there’s an aim to save vertical space, at least for the non-Gaelic Irish letters.

**Comment by georgd** [comment]
I thought of the vertical space saving argument as well and can't really support it as not much is saved. The accented uppercase letters require almost as much vertical space and the <ĥ> is really rare in Esperanto.

**Comment by georgd** [comment]
> Generally, this is more about the fact that Libertinus’ anchors on "bdhk" is not the most common one, and will seem odd to many users, regardless of the language.
> The only exception to this is probably Gaelic Irish, where it is quite common, if not more common, to have the dot above next to the ascender instead of above it.
...
> Personally, I wouldn’t favour the current design unless there’s an aim to save vertical space, at least for the non-Gaelic Irish letters.

My (Old-) Irish lessons lye far away in the past so I might be mistaken. But h with dot above doesn't make sense to me from a Gaelic perspective: The dot above marks lenition, where plosives and /m/ become fricatives and /s/ > /h/ — /h/ doesn't fit in here. Modern Irish orthography replaces the dot with a postponed <h> — that would result in word initial <hh> if there were a h with dot above.

So, I think that's no case to worry about here.

**Comment by ctrlcctrlv** [comment]
&lsaquo;ĥ&rsaquo; is not that rare, but you are right to an extent, some people have advocated replacing &lsaquo;ĥ&rsaquo; with &lsaquo;k&rsaquo;. However, &lsaquo;q&rsaquo; is also rare, and no one uses that as an excuse to make it look weird.

**Comment by ctrlcctrlv** [comment]
By the way guys, it's not really an excuse to appeal to Gaelic as a reason to maintain the ''status quo'', as of fontforge/fontforge#3645. I should know, as that was my pull request.

You can give ḣ a custom decomposition, so from:
### h + &#x25cc;&#x0307;

To
### h+ uni0307.lowanchor 

Note that all characters that will be referred to by fontforge/fontforge#3645 user decompositions must be encoded into the PUA, as only by hex can they be referred as yet due to complicated internal FontForge issue I never got around to solving. If you want, you can null the PUA char's encoding in post-processing via e.g. `ttx` or what have you (sfdLib I think).

**Comment by ctrlcctrlv** [comment]
The reason I opened this bug was I was getting ready to write a paper...

![image](https://user-images.githubusercontent.com/838783/87216102-2d3b7680-c2f1-11ea-8d37-1cb08c75322e.png)

And right when I was, something I was talking to @skef about reminded me this font exists.

But, I remembered, I'll probably want to put my Wikipedia username on it, and the ‹ĥ› is simply unacceptable, something I'd not want to turn in unless some bug that wasn't my fault and that I didn't notice caused it. (Printer...something something...PostScript...I don't know. Let's say an esoteric bug in a print-to-PDF driver where it misunderstands anchor positions, and for some reason my document wasn't using the precombined character, or it got NFD normalized along the way to the driver because some developer loved combining characters. Stranger things have happened.)

Anyway, in this discussion we've determined all speakers who've spoken up don't like it and the original author of this font gave up on it and started putting it in <del>the right</del> <del>the correct</del> an &aelig;sthetically appealing position in later fonts in the series.

Pretty much everyone seems to agree it should change now, at least for non-Gaelic. I think the only remaining issue is whether form 2 or form 3 will be used. Unfortunately for me, most of you seem to like form 3. I still think form 2 should be default, but...it is an allowable variation, I'll admit, not like form 1.

So I'd even be OK with implementing form 3 if it means form 1 goes.

(Well, except @fitojb, but they literally just seem to just be trolling at this point. Downvoting everything I write. This isn't Reddit, up boats don't matter. Look how many down boats I got [here](https://github.com/fontforge/fontforge/issues/4081) @fitojb...and yet consensus came around to my suggestion. Find something better to do with your time than clicking thumbs down on every innocuous comment. Try making an actual argument, perhaps. Or just leaving it to people who care because they actually know this language and would use this glyph.)

Mission accomplished, I suppose. Who's going to be writing the PR? I don't mind doing it if it's okay for me to take a while because I have a lot of other things to do...but I did write the feature I explained in the post above exactly for this situation, so I already know exactly how to use it. ┐(´～｀)┌

**Comment by skef** [comment]
Since I've been brought into this, I should note that I like the aesthetics of the current glyph.

**Comment by ctrlcctrlv** [comment]
Right, yes I do remember that @skef ;-)

I find it quite humorous only those who know the language prefer #2 / #3. It seems likely I'd be arguing for keeping #1 as well if I didn't know Esperanto. 😖 I wonder what causes this, if it's just "I know what it's supposed to look like and it bothers me it doesn't look like that" or what. But someone who has no long experience with the glyph, perhaps, just perhaps, they can make a more unbiased determination based on aesthetics alone and not patterns imprinted on their brain by reading and writing. 😂 

We seem to be repudiating the idea that natives are needed for type design, yikes. This train of thought is getting very politically incorrect, time for me to cut it short.

**Comment by alerque** [comment]
Let's give this discussion a rest. I'm not in a place to make a decision or fix it or even accept a contribution to fix this at the moment. Not to say I won't be, just that I'm not _yet_. I'll review the points brought up when I'm able to deal with it.

Yeh or neh votes/opinions and brief explanations of why from **other** voices are okay so I know who to follow up with later, but lets stop arguing about it and definitely stop arguing about ideology.

**Comment by ctrlcctrlv** [comment]
That sounds like a good plan to me @alerque. Thanks for the help you've been giving me with SILE, once again. I'm sorry I can be very difficult to deal with. I'm obviously not even close to the most neurotypical person in the world. Sometimes it's more obvious than at other times. This issue brought out my eccentric side, that's for sure. I do agree I've said all that should be said at this juncture.

**Comment by ctrlcctrlv** [comment]
(Sorry, just so you know, as you may not be aware, @skef and I have collaborated on FontForge on and off for over a year, so we know each other as well as you can know someone through GitHub, and nothing I wrote was intended as an attack on him, he's one of my only friends on this site. If you read it as negative, it might be because I feel like I know him a little bit so can talk a little more frankly, but this is us getting along. When we're not getting along it's way worse than this, promise.)

**Comment by Vanege** [comment]
Just to give my input as an everyday user of Esperanto from many years.

Forms:
1) Never saw it. Too weird. Looks like an error.
2) That's the normal letter.
3) Really uncommon. Does not annoy me.

**Comment by marbuljon** [comment]
I've written books etc in Esperanto. The usual problem with this letter is the circumflex is too high, which reduces or erases the white space between the text lines. This ends up very annoying and potentially confusing for visually impaired or dyslexic people, because the letter seems to merge with the letter on the line directly above it. Note that I believe the only reason we even have a standard look for this letter in Esperanto is because of how typewriters used to work, not because it's some kind of rule. Also there's really never any confusion if you replace the circumflex with another mark on Esperanto letters, it just needs to have "something" there to make it different from a normal h.

I prefer #1. When I see #2 or #3, because I'm visually impaired and the line space issue, I tend to "not see" the circumflex.

**Comment by ctrlcctrlv** [comment]
Quite right @marbuljon, that's why I wrote "closest thing to a native speaker" and not "native speaker" ;-)

This is a tangent, but many native Esperanto speakers actually don't speak that great of Esperanto, or even grammatically correct Esperanto. Here's an interesting paper about that:

Bergen, Benjamin K. [Nativization processes in L1 Esperanto](http://www.cogsci.ucsd.edu/~bkbergen/papers/NEJCL.pdf). _J. Child Lang._ **28** (2001), 575&ndash;595.

**Comment by ketsuban** [comment]
> This is a tangent, but many native Esperanto speakers actually don't speak that great of Esperanto, or even grammatically correct Esperanto.

It's odd to me that you'd assert that *denaskuloj* don't produce "good Esperanto" - I'd have thought that, being native speakers, good Esperanto is what they produce whether or not you, I or Ludwig like it. (I'm a fan of A. Z. Foreman's treatment, which sadly appears to only online as [a backup copy on the Internet Archive](https://web.archive.org/web/20171210020045/http://blogicarian.blogspot.com/2012/02/esperanto-international-auxiliary.html).)

**Comment by ctrlcctrlv** [comment]
Plenty of native English speakers don't produce good English&mdash;I scarcely believe that Esperanto is unique in this matter.

**Comment by ketsuban** [comment]
The existence of a surfeit of boorish grammarians insisting the version of English they learned or found in a book is "good English" and the sounds actually coming out of peoples' mouths are not does not prove or justify prescriptivism.

**Comment by ctrlcctrlv** [comment]
As no native speaker has commented, this argument is moot and serves no purpose.

**Comment by FloraCanou** [comment]
(1) The vertical-space argument
It's invalid. There're also capital letters with the hat that takes the same space and cannot be vertically compressed. If you don't avoid capital letters with the hat then no space can be saved. 

(2) The document argument

> I would ignore this document. It is too strange; it is in English but has German quote marks and fails to capitalize ‘Scotch’. And it is not an attempt at handsome printing. 

Totally agree. That document was printed in metal type and cannot represent digital typesetting. In particular, the selected source shows a lot of irregularities that was natural (meaning: probably not intentional) in metal typesetting but seem very coarse to contemporary eyes. 

(3) My POV
It's most preferable to place the hat just above the letter, which leads to form 2. That looks most neat and least disturbing in a running text. Also acceptable to place it just above the apex, which leads to 3, cuz that also makes logical sense. 

(4) Trivia
Prescriptivism is evil, and not necessary.

**Comment by Crissov** [comment]
If it was for me only, all lowercase letters with an ascender and a caron / háček / circumflex / hat / wedge / chevron diacritic above would learn from Czech/Slovak customs where ď, ľ and ť have the accent mark shaped like an apostrophe or acute.

**Comment by ctrlcctrlv** [comment]
This is not possible in Esperanto.

Roots can be used by supplying an apostrophe or ’.

E.g. La bird' soras

Instead of birdo...Roots of course can also end in ĥ. So we will get a collision like h'' if your method is used.

**Comment by alerque** [comment]
Why the closure? I do plan on doing *something* about this, at the least an alternate.

**Comment by ctrlcctrlv** [comment]
GitHub won't let me delete the issue. I would prefer you do that and open another one. I'm tired of being notified and also being laughed at.† Long since ready to move on. 

† Recognizing part of the reason that's happening is my own fault

**Comment by alerque** [comment]
I believe you can use the Unsubscribe link on the right no stop getting notified, no? If not I can lock this and open a new one under my name. Actually I think until I go to actually implement this we've collected all the feedback I need so I'll just lock this frequently off topic chain until further notice. If somebody has their own brand of concerns or ideas they can use their own issues ;-)

**Comment by alerque** [comment]
More info in this Twitter thread:

> Question for type twitter: In 'hcircumflex', what is the correct position for the circumflex? I have always assumed it should be on the top of the ascender, but some typefaces push it to the middle of the letter. https://t.co/KF973q81Pw
>
> https://twitter.com/kaja_slojewska/status/1328490359630950402
